docs/plugin-protocol/tfplugin5.proto

```diff
@@ -811,3 +819,45 @@ message CallFunction {
         FunctionError error = 2;
     }
 }
+
+message ListResource {
+    message Request {
+        // type_name is the list resource type name.
+        string type_name = 1;
+
+        // configuration is the list ConfigSchema-based configuration data.
+        DynamicValue config = 2;
+    }
+
+    message Event {
+        // Resource contains all data associated with each resource instance.
+        message Result {
+            // identity is the resource identity data of the resource instance.
+            ResourceIdentityData identity = 1;
+
+            // display_name can be displayed in a UI to make it easier for humans to identify a resource
+            string display_name = 2;
+
+            // optional resource object which can be useful when combining list blocks in configuration
+            DynamicValue resource_object = 3;
+        }
+
+        oneof response {
```

Comment:
I don't know if we want `oneof` here, doesn't that preclude returning both? Terraform has typically assumed there will always be both a result and a diagnostic in every response. While `ListResource` may not require a result with an error like `ApplyResourceChange` does, there may still be warnings included with a valid result.

Reply:
yeah, returning a warning in combination with a result won't be possible with `oneof` 🤔 

We could flatten the `Event` message to something like this:
```proto
message Event {
    // identity is the resource identity data of the resource instance.
    ResourceIdentityData identity = 1;

    // display_name can be displayed in a UI to make it easier for humans to identify a resource
    string display_name = 2;

    // optional resource object which can be useful when combining list blocks in configuration
    optional DynamicValue resource_object = 3;

    // A warning or error diagnostics for this event
    repeated Diagnostic diagnostic = 4;
}
```

This means that we cannot return a single error without the additional fields (all being null?). But maybe that's OK?

Reply:
> This means that we cannot return a single error without the additional fields 

This is what is typical for other RPCs, so I would also assume it is okay.